server.go

```diff
@@ -258,7 +256,7 @@ func handlePacket(s *Server, p orderedRequest) error {
 		f, ok := s.getHandle(p.Handle)
 		if ok {
 			err = nil
-			data := make([]byte, clamp(p.Len, s.maxTxPacket))
+			data := p.getDataSlice()
```

Comment:
It might be worth mentioning down below at the `Data:` field that we need to _not_ clamp the capacity, as we use it later in the marshalling. e.g. `// do not use data[:n:n] here to clamp the capacity, we allocated extra capacity above to avoid reallocations`